File: footnest-backend/src/main/java/com/footnest/footnest_backend/entity/MatchStatistics.java
package com.footnest.footnest_backend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(
    name = "match_statistics",
    uniqueConstraints = {
        @UniqueConstraint(
            columnNames = {"match_id", "team_id"}
        )
    }
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MatchStatistics {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "match_id", nullable = false)
    private FootballMatch match;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id", nullable = false)
    private Team team;


    private Double xg;

    private Double possession;

    private Integer totalShots;

    private Integer shotsOnTarget;

    private Integer bigChances;

    private Integer corners;

    private Integer yellowCards;

    private Integer redCards;

    private Integer fouls;
}

File: footnest_app/lib/models/match/match_statistics.dart
class MatchStatistics {

  final int id;

  final int teamId;
  final String teamName;
  final String teamLogo;

  final double? xg;
  final double? possession;

  final int? totalShots;
  final int? shotsOnTarget;
  final int? bigChances;
  final int? corners;

  final int? yellowCards;
  final int? redCards;
  final int? fouls;


  MatchStatistics({
    required this.id,
    required this.teamId,
    required this.teamName,
    required this.teamLogo,
    this.xg,
    this.possession,
    this.totalShots,
    this.shotsOnTarget,
    this.bigChances,
    this.corners,
    this.yellowCards,
    this.redCards,
    this.fouls,
  });


  factory MatchStatistics.fromJson(
      Map<String,dynamic> json
  ) {

    return MatchStatistics(
      id: json["id"],
      teamId: json["teamId"],
      teamName: json["teamName"],
      teamLogo: json["teamLogo"],
      xg: json["xg"],
      possession: json["possession"],
      totalShots: json["totalShots"],
      shotsOnTarget: json["shotsOnTarget"],
      bigChances: json["bigChances"],
      corners: json["corners"],
      yellowCards: json["yellowCards"],
      redCards: json["redCards"],
      fouls: json["fouls"],
    );

  }

}